pyro/distributions/log_normal.py

```diff
@@ -32,40 +32,23 @@ def __init__(self, mu, sigma, *args, **kwargs):
             raise ValueError("Expected mu.size() == sigma.size(), but got {} vs {}".format(mu.size(), sigma.size()))
         super(LogNormal, self).__init__(*args, **kwargs)
 
-    def batch_shape(self, x=None):
-        event_dim = 1
-        mu = self.mu
-        if x is not None:
-            if x.size()[-event_dim] != mu.size()[-event_dim]:
-                raise ValueError("The event size for the data and distribution parameters must match.\n"
-                                 "Expected x.size()[-1] == self.mu.size()[-1], but got {} vs {}".format(
-                                     x.size(-1), mu.size(-1)))
-            try:
-                mu = self.mu.expand_as(x)
-            except RuntimeError as e:
-                raise ValueError("Parameter `mu` with shape {} is not broadcastable to "
-                                 "the data shape {}. \nError: {}".format(mu.size(), x.size(), str(e)))
-        return mu.size()[:-event_dim]
+    def batch_shape(self):
+        return self.mu.size()
 
     def event_shape(self):
-        event_dim = 1
-        return self.mu.size()[-event_dim:]
+        return torch.Size()
 
     def sample(self, sample_shape=torch.Size()):
         shape = sample_shape + self.mu.size()
         eps = Variable(torch.randn(shape).type_as(self.mu.data))
         z = self.mu + self.sigma * eps
         return torch.exp(z)
 
-    def batch_log_pdf(self, x):
-        mu = self.mu.expand(self.shape(x))
-        sigma = self.sigma.expand(self.shape(x))
-        ll_1 = Variable(torch.Tensor([-0.5 * np.log(2.0 * np.pi)]).type_as(mu.data).expand_as(x))
-        ll_2 = -torch.log(sigma * x)
-        ll_3 = -0.5 * torch.pow((torch.log(x) - mu) / sigma, 2.0)
-        batch_log_pdf = torch.sum(ll_1 + ll_2 + ll_3, -1)
-        batch_log_pdf_shape = self.batch_shape(x) + (1,)
-        return batch_log_pdf.contiguous().view(batch_log_pdf_shape)
+    def log_prob(self, x):
+        ll_1 = Variable(torch.Tensor([-0.5 * np.log(2.0 * np.pi)]).type_as(self.mu.data).expand_as(x))
```

Comment:
This simpler version works correctly on multi-GPU systems:
```py
ll_1 = -0.5 * np.log(2.0 * np.pi)
```
It's fine to address this in a later PR though.

Reply:
Oddly, enough this threw an error `AttributeError: type object 'Variable' has no attribute 'from_numpy'` when we add this to a Variable later, but changing to python numeric type works.

Reply:
Oh right, I've had that problem too.